packages/frontend-shared/cypress/e2e/e2ePluginSetup.ts

```diff
@@ -159,70 +161,77 @@ async function makeE2ETasks () {
 
       const operationCount: Record<string, number> = {}
 
-      sinon.stub(ctx.util, 'fetch').get(() => {
-        return async (url: RequestInfo, init?: RequestInit) => {
-          if (String(url).endsWith('/test-runner-graphql')) {
-            const { query, variables } = JSON.parse(String(init?.body))
-            const document = parse(query)
-            const operationName = getOperationName(document)
-
-            operationCount[operationName ?? 'unknown'] = operationCount[operationName ?? 'unknown'] ?? 0
-
-            let result = await execute({
-              operationName,
-              document,
-              variableValues: variables,
-              schema: cloudSchema,
-              rootValue: CloudRunQuery,
-              contextValue: {
-                __server__: ctx,
-              },
-            })
-
-            operationCount[operationName ?? 'unknown']++
-
-            if (remoteGraphQLIntercept) {
-              try {
-                result = await remoteGraphQLIntercept({
-                  operationName,
-                  variables,
-                  document,
-                  query,
-                  result,
-                  callCount: operationCount[operationName ?? 'unknown'],
-                }, testState)
-              } catch (e) {
-                const err = e as Error
-
-                result = { data: null, extensions: [], errors: [new GraphQLError(err.message, undefined, undefined, undefined, undefined, err)] }
-              }
-            }
+      sinon.stub(ctx.util, 'fetch').callsFake(async (url: RequestInfo, init?: RequestInit) => {
```

Comment:
Was having trouble following when/how this was mocked in different situations when working through the e2e test for ensuring the app starts up without delay. Making it a method made it easier to track / ensured the getter wasn't accessed & cached.

Reply:
This part of the code is pretty complex - it looks like it mostly exists for the purpose of just 2-3 tests, I wonder if we can change those tests so we don't even need this anymore.

Reply:
> This part of the code is pretty complex - it looks like it mostly exists for the purpose of just 2-3 tests, I wonder if we can change those tests so we don't even need this anymore.

Not sure what "it" is referring to here. This code exists as the default handler for all remote requests in the tests. We've talked about creating a server module to handle this, but that'd probably be a separate task we'd want to design the requirements around.

Reply:
Oh, realizing you're probably referring to the remote fetch stuff specifically. Yeah, agreed, seeing if I can extract that to just the tests that need it.

Reply:
Was referring the logic around stubbing `fetch` and specifically [this part](https://github.com/cypress-io/cypress/blob/10.0-release/packages/frontend-shared/cypress/e2e/e2ePluginSetup.ts#L169) where we count the number of operations - I thought that was what you were referring to w.r.t a lot of complexity to grok.

It looks like only a handful of tests (cloud page) cares about the operation count - it's quite a bit of code for 1-2 tests. Now I'm thinking I may have misunderstood your comment.

Maybe this stub is doing far more than I'm realizing.

Reply:
Oh my original comment was about the fact that I changed from `get` to `callsFake` which caused this big indent diff

Reply:
Yeah, the code is really different from when I originally commented. I'm just going to mark this as resolved.